src/main/scala/org/apache/spark/mllib/optimization/tfocs/SmoothFunction.scala

```diff
@@ -72,3 +69,70 @@ class SquaredErrorRDDVector(x0: RDD[Vector]) extends SmoothFunction[RDD[Vector]]
   }
 }
 
+/**
+ * The huber loss function applied to RDD[Vector] vectors.
+ *
+ * @param x0 The vector against which loss should be computed.
+ * @param tau The huber loss parameter.
+ */
+class SmoothHuberRDDVector(x0: RDD[Vector], tau: Double)
+    extends SmoothFunction[RDD[Vector]] with Serializable {
+
+  x0.cache()
+
+  override def apply(x: RDD[Vector], mode: Mode): Value[RDD[Vector]] = {
+
+    val diff = x.diff(x0)
+    if (mode.f && mode.g) diff.cache()
+
+    val f = if (mode.f) {
+      Some(diff.mapElements(y =>
+        if (math.abs(y) <= tau) 0.5 * y * y / tau else math.abs(y) - tau / 2.0).sum)
```

Comment:
I can convert this and the corresponding example in the logistic implementation to do an aggregate instead of map + sum.